Resolve `myst.xref_missing` suppression in sphinx config file

~We should remove `ref.myst` suppression from Sphinx's `conf.py` in our docs once we update MyST parser to >=0.19.0.~

~https://myst-parser.readthedocs.io/en/v0.19.0/develop/_changelog.html#extended-markdown-links~

Edit: we need to resolve invalid cross-references in order to remove `myst.xref_missing` suppression in sphinx config file

```console
$ make dirhtml
...
Warning, treated as error:
/go/github.com/scylladb/scylla-operator/docs/source/contributing.md:117:'myst' cross-reference target not found: 'updating-your-fork' [myst.xref_missing]
make: *** [Makefile:37: dirhtml] Error 2
```

--- a/docs/source/_ext/scylladb_markdown.py
+++ b/docs/source/_ext/scylladb_markdown.py
@@ -27,10 +27,10 @@ class MystParser(BaseParser):
         try:
             self.app.setup_extension('myst_parser')
             self.app.config.myst_enable_extensions = ["colon_fence"]
+            self.app.config.myst_heading_anchors = 6
 
-            # TODO: https://github.com/scylladb/scylla-operator/issues/1421
             # TODO: https://github.com/scylladb/scylla-operator/issues/1422
-            self.app.config.suppress_warnings = ["myst.xref_missing", "myst.header"]
+            self.app.config.suppress_warnings = ["myst.header"]
 
         except ImportError:
             raise RuntimeError("myst-parser is not installed")

--- a/docs/source/eks.md
+++ b/docs/source/eks.md
@@ -20,7 +20,7 @@ cd examples/eks
 ./eks.sh -z "$EKS_ZONES" -r "$EKS_REGION"
 ```
 
-After you deploy, see how you can [benchmark your cluster with cassandra-stress](#benchmark-with-cassandra-stress).
+After you deploy, see how you can [benchmark your cluster with cassandra-stress](generic.md#benchmark-with-cassandra-stress).
 
 ## Walkthrough
 

--- a/docs/source/gke.md
+++ b/docs/source/gke.md
@@ -27,7 +27,7 @@ cd examples/gke
 Make sure to pass a ZONE (ex.: us-west1-b) and not a REGION (ex.: us-west1) or it will deploy nodes in each ZONE available in the region.
 :::
 
-After you deploy, see how you can [benchmark your cluster with cassandra-stress](#benchmark-with-cassandra-stress).
+After you deploy, see how you can [benchmark your cluster with cassandra-stress](generic.md#benchmark-with-cassandra-stress).
 
 ## Walkthrough
 

--- a/docs/source/multidc/multidc.md
+++ b/docs/source/multidc/multidc.md
@@ -598,4 +598,4 @@ kubectl --context="${CONTEXT_DC2}" patch scyllacluster/scylla-cluster --type mer
 ## ScyllaDBMonitoring
 
 To monitor your cluster, deploy ScyllaDBMonitoring in every datacenter independently.
-To deploy ScyllaDB Monitoring, follow the steps described in [Deploy managed monitoring](../monitoring#deploy-managed-monitoring) in ScyllaDB Operator documentation.
+To deploy ScyllaDB Monitoring, follow the steps described in [Deploy managed monitoring](../monitoring.md#deploy-managed-monitoring) in ScyllaDB Operator documentation.

